lib/chained_job/process.rb

```diff
@@ -22,13 +22,22 @@ def run
       with_hooks do
         return finished_worker unless argument
 
-        job_instance.process(argument)
+        begin
+          job_instance.process(argument)
+        rescue StandardError => e
+          push_job_arguments_back if handle_retry?
+          raise e
+        end
         job_instance.class.perform_later(args, worker_id, job_tag)
       end
     end
 
     private
 
+    def handle_retry?
+      job_instance.methods.include?('handle_retry') && job_instance.handle_retry
```

Comment:
```suggestion
      job_instance.try(:handle_retry)
```

This could be enough.

Reply:
@tomasv Good point, this works. Before that, I tried &., but .try works even with non-declared methods too. 

Tests and code updated.